js/src/hooks/usePreferences.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * External dependencies
+ */
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as preferencesStore } from '@wordpress/preferences';
+
+const STORE_NAME = 'woocommerce/google-listings-and-ads';
+
+/**
+ * A custom hook for managing preferences in the 'woocommerce/google-listings-and-ads' scope.
+ *
+ * @return {Object} An object with `get` and `set` methods for preferences.
+ */
+export const usePreferences = () => {
+	const { set } = useDispatch( preferencesStore );
+	const { get } = useSelect( ( select ) => ( {
```

Comment:
@ankitrox There's an issue with this approach since you're just getting a reference to the selector and if the value changes, the component will not be re rendered. Maybe it'll be easier to just have a `usePreference` hook which returns a value.